app/src/main/java/com/wtmberlin/data/Repository.kt

```diff
@@ -1,67 +1,56 @@
 package com.wtmberlin.data
 
-import io.reactivex.Flowable
-import io.reactivex.Single
-
-open class Repository(private val apiService: ApiService, private val database: Database) {
-    fun venues(): Flowable<Result<List<VenueName>>> {
-        return apiService.events()
-            .map { it.map(WtmEvent::toVenueName) }
-            .map { it.distinct() }
-            .map { Result(loading = false, data = it, error = null) }
-            .onErrorReturn { Result(loading = false, data = null, error = it) }.toFlowable()
+import kotlinx.coroutines.withContext
+
+open class Repository(
+    private val apiService: ApiService,
+    private val database: Database,
+    private val dispatchers: CoroutinesDispatcherProvider
+) {
+    suspend fun venues() = withContext(dispatchers.io) {
+        com.wtmberlin.data.runCatching {
+            Result(
+                loading = true, data = apiService.events()
+                    .map(::toVenueName)
+                    .distinct()
+                    .sortedBy { it.name.toLowerCase().trimStart() }, error = null
+            )
+        }
     }
 
-
-    open fun events(): Flowable<Result<List<WtmEvent>>> {
-        return eventsResource.values()
-            .map {
-                if (it.data != null && it.data.isEmpty()) {
-                    Result(it.loading, null, it.error)
-                } else {
-                    it
-                }
-            }
-            .doOnSubscribe { eventsResource.refresh() }
+    suspend fun events() = withContext(dispatchers.io) {
+        fetchEvents()
     }
 
-    fun refreshEvents() {
-        eventsResource.refresh()
+    suspend fun refreshEvents() = withContext(dispatchers.io) {
```

Comment:
You can simplify it by doing this
```suggestion
    suspend fun refreshEvents() = withContext(dispatchers.io, ::fetchEvents)
```